### PR TITLE
Everywhere: Update code to compile under zig 0.11.0

### PR DESCRIPTION
**Pull request by hardliner66**
I updated the code to compile with the new 0.11.0 release. This way there is a known stable version to compile the code or work on bugs.

I also changed the commit hash to the hash of the 0.11.0 release and noted that it compiles with the official 0.11.0 compiler binary.

Most of these changes were done automatically by the vscode plugin and could probably be simplified by removing the explicit `@as` cast, which is something I'm not really sure you want, so I left those in there.

I ran all the tests and everything seems to pass.

### CODE REVIEW

**Review by sin-ack — CHANGES_REQUESTED** [review]
Could you also update the CI to use 0.11.0 please? I'll give a more thorough review later today.

**Comment by hardliner66** [comment]
> Could you also update the CI to use 0.11.0 please? I'll give a more thorough review later today.

I updated the zig version in the CI and tested it on my github:
https://github.com/hardliner66/zigself/actions

**Comment by sin-ack** [comment]
Thanks! Seems like the tests pass now. I'll give a more thorough review later as I said, however at a glance, it would be great if we could remove `@as` casts whenever possible since it nullifies one of the primary goals of the type removal (automatic casting to prevent missed cast type change errors).

**Comment by hardliner66** [comment]
> Thanks! Seems like the tests pass now. I'll give a more thorough review later as I said, however at a glance, it would be great if we could remove `@as` casts whenever possible since it nullifies one of the primary goals of the type removal (automatic casting to prevent missed cast type change errors).

Sure. I already started doing this yesterday, but thought it would be better to ask you first on how you wanted the code to look. I'll update the PR once I'm done.

**Comment by hardliner66** [comment]
I removed each `@as` that was easily doable and pulled out some of the cases into temporary variables, so that the `@as` wasn't needed anymore.

File: src/runtime/bytecode/AstGen.zig [hunk 1]
```diff
@@ -146,8 +147,9 @@ fn generateBlock(self: *AstGen, executable: *Executable, block: *Block, block_no
 
     const block_index = try self.generateSlotsAndCodeCommon(executable, block, block_node.slots, block_node.statements.value.statements);
     const block_location = self.allocateRegister();
+    const slot_count: u32 = @intCast(block_node.slots.len);
     try block.addInstruction(executable.allocator, .CreateBlock, block_location, .{
-        .slot_count = @intCast(u32, block_node.slots.len),
+        .slot_count = slot_count,
```
Comment (sin-ack) [inline src/runtime/bytecode/AstGen.zig:152]: I remember having a problem with these when I attempted to do an `intCast` update myself. Do you know whether `@intCast` still fails to infer the type for generic functions here? I'd rather have this do a `@intCast` directly into the struct. (not a blocker)

File: src/runtime/bytecode/astcode/register_location.zig [hunk 2]
```diff
@@ -27,11 +27,11 @@ pub const RegisterLocation = enum(u32) {
     }
 
     pub fn fromIndexAllowNil(index: u32) RegisterLocation {
-        return @intToEnum(RegisterLocation, index);
+        return @as(RegisterLocation, @enumFromInt(index));
```
Comment (sin-ack) [inline src/runtime/bytecode/astcode/register_location.zig]: Does this fail to infer the type?

File: src/runtime/objects/map.zig [hunk 3]
```diff
@@ -168,9 +172,9 @@ pub const Map = extern struct {
         const size = requiredSizeForAllocatingMapMap();
         var memory_area = token.allocate(.Object, size);
 
-        var map_map = @ptrCast(Map.Ptr, memory_area);
+        var map_map: Map.Ptr = @ptrCast(memory_area);
         // A defined undefined value.
-        map_map.init(.MapMap, @intToPtr(Map.Ptr, 0x13370));
+        map_map.init(.MapMap, @as(Map.Ptr, @ptrFromInt(0x13370)));
```
Comment (sin-ack) [inline src/runtime/objects/map.zig]: Does this not infer either? That's weird.

**Review by sin-ack — COMMENTED** [review]
Awesome work! Thank you. I only have a couple questions before this can be merged.

Also, please put the category prefix "Everywhere" before the commit title.

File: src/runtime/bytecode/AstGen.zig (hunk 1 above)
Comment (hardliner66) [reply src/runtime/bytecode/AstGen.zig:152]: Yes, it still fails. That's why I moved it to a separate line.

File: src/runtime/bytecode/astcode/register_location.zig (hunk 2 above)
Comment (hardliner66) [reply src/runtime/bytecode/astcode/register_location.zig]: Good catch, I only fixed the casts. I'm gonna fix those cases as well.

File: src/runtime/objects/map.zig (hunk 3 above)
Comment (hardliner66) [reply src/runtime/objects/map.zig]: Same as above.

Comment (hardliner66) [reply src/runtime/objects/map.zig]: I updated all cases where a cast in the form `XFromY` was used in conjunction with `@as`.

**Comment by hardliner66** [comment]
I updated all cases where a cast in the form `XFromY` was used in conjunction with `@as` and wasn't needed anymore.

**Comment by sin-ack** [comment]
Ah, I meant commit message, not PR title :sweat_smile: If you can do that real quick I'll merge this.

**Comment by hardliner66** [comment]
I changed the commit message :D

**Review by sin-ack — APPROVED** [review]
Thanks a lot! Great work.

**Pull request merged**